cmd/disk-cache/core/part_consumer.go

```diff
@@ -31,11 +34,20 @@ func (m *partitionConsumer) Meta() *dstk.Partition {
 
 // thread safe
 func (m *partitionConsumer) get(req *dstk.DcGetReq) (interface{}, error) {
-	return m.pc.Get(req.GetKey())
+	document, err := m.pc.Get(req.GetKey())
+	if err == badger.ErrKeyNotFound {
+		return nil, core.ErrKeyAbsent(req.GetKey()).Err()
+	}
+	return document, err
 }
 
 func (m *partitionConsumer) put(req *dstk.DcPutReq) (interface{}, error) {
-	return nil, m.pc.Put(req.GetKey(), req.GetValue(), req.GetTtlSeconds())
+	document := &dstk.DcDocument{
+		Value: req.GetValue(),
+		Etag: req.GetEtag(),
+		LastUpdatedEpochSeconds: time.Now().Unix(),
```

Comment:
dont use `time.Now()`. this makes the system hard to reproduce and test. @harshavardhangelivi has implemented some wrapper class with ideas from https://quii.gitbook.io/learn-go-with-tests/go-fundamentals/mocking

Reply:
Done.